src/x86_64/vdso.c

```diff
@@ -0,0 +1,142 @@
+/* VDSO syscall implementations
+ *
+ * Unless there's a BUG somewhere, or Linux implements some new syscalls here,
+ * there shouldn't be any reason to modify this
+ */
+
+#include <runtime.h>
+#include <unix_internal.h>
+#include <vdso.h>
+
+static sysreturn
+fallback_clock_gettime(clockid_t clk_id, struct timespec * tp)
+{
+    return do_syscall(SYS_clock_gettime, clk_id, tp);
+}
+
+static sysreturn
+fallback_gettimeofday(struct timeval * tv, void * tz)
+{
+    return do_syscall(SYS_gettimeofday, tv, tz);
+}
+
+static sysreturn
+fallback_time(time_t * t)
+{
+    return do_syscall(SYS_time, t, 0);
+}
+
+static sysreturn
+do_vdso_clock_gettime(clockid_t clk_id, struct timespec * tp)
+{
+    timestamp ts;
+
+    switch (clk_id) {
```

Comment:
Not clear to my why there is a switch here on clk_id when vdso_now does so also; could it be confined to vdso_now()?